# Question

Is this function safe? Why or why not?

```js
const getUser = async (userId) => {
  const url = `https://jsonplaceholder.typicode.com/users/${userId}`;
  const response = await fetch(url);
  const users = await response.json();
  console.log(users);
  return users;
};
getUser(-1);
```

Again, running the code gives you the answer, so try to figure it out on your own first!

# Answer

I think this function might be safe because there is string interpolation to kind of hide what was looked up. However, there are some missing parts. `response` is not returned, so the URL is not fetched for the next line, which should be checking if `response` is ok. If the response is ok, then `response.json()` should be returned. Error handling (`catch`) is also missing. Lastly, I do not think an id of -1 is possible.

# Teacher Feedback

You are correct, there is no error handling so this fetch call is not safe, it will run but will not return an error!

Score: 3/3